Crash when signature with invalid JWK is attempted

If an invalid JWK that is missing the private exponent `p` is passed to `pkey.new()` for the `OKP` type, the subsequent call to `pkey:sign()` will result in a crash and kill the worker.

Can be reproduced with:
```
local pkey, err = openssl_pkey.new(
  '{"kty" : "OKP", "crv" : "Ed25519","x" : "MKBCTNIcKUSDii11ySs3526iDZ8AiTo7Tu6KPAqv7D4"}',
  {format = "JWK"}
)
pkey:sign("something")
```
Using the following JWK instead works as expected: `{"kty" : "OKP", "crv" : "Ed25519","x" : "MKBCTNIcKUSDii11ySs3526iDZ8AiTo7Tu6KPAqv7D4", "d" : "4Etl6SRW2YiLUrN5vfvVHuhp7x8PxltmWWlbbM4IFyM"}`.

Thanks @samugi , seems to be an bug on openssl, there's no guard before calling sign https://github.com/openssl/openssl/blob/OpenSSL_1_1_1r/crypto/ec/curve25519.c#L5442
then ultimately fail in
https://github.com/openssl/openssl/blob/OpenSSL_1_1_1r/crypto/sha/sha512.c#L307

note in stacktrace `private_key` (a char[32] is NULL):
```
#0  SHA512_Update (c=c@entry=0x7fffffffd600, _data=<optimized out>, len=len@entry=32) at crypto/sha/sha512.c:307
#1  0x00007ffff730c565 in ED25519_sign (out_sig=out_sig@entry=0x7ffff7ea6720 "", message=message@entry=0x7ffff7eec948 "something",
    message_len=message_len@entry=9, public_key=0x5555558cb450 "}\243\216\067|\341k(t\005\207q\300N\363>#Gp`d\251\252\332I[\263\223. \fk",
    private_key=<optimized out>) at crypto/ec/curve25519.c:5442
#2  0x00007ffff735b281 in pkey_ecd_digestsign25519 (ctx=<optimized out>, sig=0x7ffff7ea6720 "", siglen=0x7ffff7ea6d80, tbs=0x7ffff7eec948 "something",
    tbslen=9) at crypto/ec/ecx_meth.c:753
#3  0x00007ffff7c0d135 in lj_vm_ffi_call () from /usr/local/openresty/luajit/lib/libluajit-5.1.so.2
...
```

let me submit an issue at upstream. for now i'm going to do a pre-check on if pkey is private key before sign.
other keys like EC, RSA will fail like `pkey:sign: EVP_DigestSign: ec/ecdsa_ossl.c:174:error:100F907D:elliptic curve routines:ossl_ecdsa_sign_sig:missing private key`

--- a/lib/resty/openssl/pkey.lua
+++ b/lib/resty/openssl/pkey.lua
@@ -658,6 +658,11 @@ function _M:decrypt(s, padding)
 end
 
 function _M:sign_raw(s, padding)
+  -- TODO: temporary hack before OpenSSL has proper check for existence of private key
+  if self.key_type_is_ecx and not self:is_private() then
+    return nil, "pkey:sign_raw: missing private key"
+  end
+
   return asymmetric_routine(self, s, ASYMMETRIC_OP_SIGN_RAW, padding)
 end
 
@@ -719,6 +724,11 @@ local function sign_verify_prepare(self, fint, md_alg, padding, opts)
 end
 
 function _M:sign(digest, md_alg, padding, opts)
+  -- TODO: temporary hack before OpenSSL has proper check for existence of private key
+  if self.key_type_is_ecx and not self:is_private() then
+    return nil, "pkey:sign: missing private key"
+  end
+
   if digest_lib.istype(digest) then
     local length = ptr_of_uint()
     if C.EVP_SignFinal(digest.ctx, self.buf, length, self.ctx) ~= 1 then

